### ISSUE TITLE
Notice

### ISSUE THREAD

**Issue by marcelloh** [wapmorgan/PhpCodeFixer#12]
While running your code against mine
Notice: Uninitialized string offset: 1 in C:\data\QualityReport\vendor\wapmorgan\php-code-fixer\src\PhpCodeFixer.php on line 177

**Comment by marcelloh** [comment]
I debugged a bit:
The notice is on traits

**Comment by wapmorgan** [comment]
Can you post this trait here?

**Comment by marcelloh** [comment]
```
<?php

namespace Illuminate\Support\Traits;

use Closure;
use BadMethodCallException;

trait Macroable
{
    /**
     * The registered string macros.
     *
     * @var array
     */
    protected static $macros = [];

    /**
     * Register a custom macro.
     *
     * @param  string    $name
     * @param  callable  $macro
     * @return void
     */
    public static function macro($name, callable $macro)
    {
        static::$macros[$name] = $macro;
    }

    /**
     * Checks if macro is registered.
     *
     * @param  string  $name
     * @return bool
     */
    public static function hasMacro($name)
    {
        return isset(static::$macros[$name]);
    }

    /**
     * Dynamically handle calls to the class.
     *
     * @param  string  $method
     * @param  array   $parameters
     * @return mixed
     *
     * @throws \BadMethodCallException
     */
    public static function __callStatic($method, $parameters)
    {
        if (! static::hasMacro($method)) {
            throw new BadMethodCallException("Method {$method} does not exist.");
        }

        if (static::$macros[$method] instanceof Closure) {
            return call_user_func_array(Closure::bind(static::$macros[$method], null, static::class), $parameters);
        }

        return call_user_func_array(static::$macros[$method], $parameters);
    }

    /**
     * Dynamically handle calls to the class.
     *
     * @param  string  $method
     * @param  array   $parameters
     * @return mixed
     *
     * @throws \BadMethodCallException
     */
    public function __call($method, $parameters)
    {
        if (! static::hasMacro($method)) {
            throw new BadMethodCallException("Method {$method} does not exist.");
        }

        if (static::$macros[$method] instanceof Closure) {
            return call_user_func_array(static::$macros[$method]->bindTo($this, static::class), $parameters);
        }

        return call_user_func_array(static::$macros[$method], $parameters);
    }
}

```

**Comment by wapmorgan** [comment]
Thanks for feedback! Fixed in **2.0.5**. Please, upgrade and test your code.

**Comment by marcelloh** [comment]
I finally got the time to test, and you fixed it :-)
